Not return `err` when pass `random str or invalid privateKey` to `func PrivateKeyFromBase58`

I just tried to use the `func` to `validate private key` but got some other behavior that `func`.

https://github.com/gagliardetto/solana-go/blob/main/keys.go#L48

It's correct behavior ?

Also I want to add some notes if pass `incorrect value or random str` the `fucn` and call public method `PublicKey()` get `error`.

```sh
panic: runtime error: slice bounds out of range [x:x]
```

If pass correct 'privateKey', everything will be fine and return `publicKey`.

--- a/keys_test.go
+++ b/keys_test.go
@@ -144,6 +144,41 @@ func TestPrivateKeyFromSolanaKeygenFile(t *testing.T) {
 	}
 }
 
+// TestPrivateKeyFromBase58RejectsWrongSizedInput pins that arbitrary base58
+// strings whose decoded length is not 64 bytes are rejected with the
+// "invalid private key size" error. Issue #232 reported that
+// PrivateKeyFromBase58 silently accepted random/invalid input; the
+// size-check path is the first line of defense and is exercised here.
+// Sibling tests below cover the seed/public-key mismatch and off-curve paths.
+func TestPrivateKeyFromBase58RejectsWrongSizedInput(t *testing.T) {
+	cases := []struct {
+		name  string
+		input string
+	}{
+		// "abc" decodes to 2 bytes via base58 — far below 64.
+		{"short_base58", "abc"},
+		// "1111111111" decodes to 9 zero bytes — also far below 64.
+		{"all_ones_short", "1111111111"},
+		// A 32-byte all-ones base58 string decodes to roughly 22 bytes,
+		// still nowhere near the 64-byte ed25519 private-key size.
+		{"medium_random", "ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ"},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := PrivateKeyFromBase58(tc.input)
+			require.Error(t, err, "expected error for %q", tc.input)
+			require.Contains(
+				t,
+				err.Error(),
+				"invalid private key size",
+				"expected size-mismatch error, got: %v",
+				err,
+			)
+		})
+	}
+}
+
 func TestPrivateKeyFromBase58RejectsMismatchedSeedAndPublicKey(t *testing.T) {
 	original := MustPrivateKeyFromBase58("66cDvko73yAf8LYvFMM3r8vF5vJtkk7JKMgEKwkmBC86oHdq41C7i1a2vS3zE1yCcdLLk6VUatUb32ZzVjSBXtRs")
 	require.Len(t, original, PrivateKeyLength)

